web/static/js/components/simulation/SimulationSteps.jsx

```diff
@@ -0,0 +1,139 @@
+// @flow
+import React, { Component } from 'react'
+import { icon } from '../../step'
+import { translate } from 'react-i18next'
+import { Card, UntitledIfEmpty } from '../ui'
+import classNames from 'classnames/bind'
+import { flatMapDepth } from 'lodash'
+import propsAreEqual from '../../propsAreEqual'
+
+type Step = {
+  type: string,
+  title: string,
+  id: string,
+  steps: ?Array<Step>
+}
+
+type Submission = {
+  stepId: string,
+  response: ?string
+}
+
+type Props = {
+  submissions: Array<Submission>,
+  steps: Array<Step>,
+  currentStepId: string,
+  t: Function
+}
+
+type StepResponses = {
+  [stepId: string]: string
+}
+
+type State = {
+  headedBySectionSteps: Array<Step>,
+  stepResponses: StepResponses
+}
+
+const SimulationSteps = translate()(class extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = this.stateFromProps(props)
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (propsAreEqual(this.props, newProps)) return
+    this.setState(this.stateFromProps(newProps))
+  }
+
+  stateFromProps(props) {
+    const { steps, submissions } = props
+    const headedBySectionSteps = flatMapDepth(steps, step => {
+      if (step.type == 'section') {
+        return [
+          step,
+          step.steps
+        ]
+      } else {
+        return step
+      }
+    }, 2)
+    const stepResponses = submissions.reduce(
+      (stepResponses, submission) => {
+        if (submission.response) stepResponses[submission.stepId] = submission.response
+        return stepResponses
+      },
+      {}
+    )
+    return {
+      headedBySectionSteps,
+      stepResponses
+    }
+  }
+
+  isActive = (stepId: string): boolean => {
+    const { currentStepId } = this.props
+    return currentStepId == stepId
+  }
+
+  isCompleted = (stepId: string): boolean => {
+    if (this.isActive(stepId)) return false
+    const { submissions } = this.props
+    return submissions.some(submission => submission.stepId == stepId)
+  }
+
+  wasSkipped = (stepId: string): boolean => {
+    if (this.isActive(stepId) || this.isCompleted(stepId)) return false
+    const { currentStepId } = this.props
+    if (currentStepId) {
+      const { headedBySectionSteps } = this.state
+      return headedBySectionSteps.findIndex(step => step.id == stepId) < headedBySectionSteps.findIndex(step => step.id == currentStepId)
+    } else {
+      return true
```

Comment:
Why if there is no current step all steps are marked as "skipped"?

The section headers are also marked as completed or they render different that question steps.

Reply:
The truth is that the case I'm covering defaulting to `true` here is this (simulation ended)

![simulation ended](https://user-images.githubusercontent.com/39921597/82402578-2a2fb280-9a33-11ea-9fb7-bf273dc8fb6f.png)

I'm not really sure if defaulting it to `true` is the best solution. What do you think?

Regarding the section headers, they are always being classified as skipped (because of the above default). I think visually they can be OK, but I'm pretty sure the semantic isn't good. I'll try to make it better and let you know if I find that I can't.

Reply:
By default questions should not be marked as skipped.

Even if visually they look the same. For now it's fine if by default pending and skipped questions look the same, but let's not add a css class of skipped to default so we can iterate later on the styling if needed.

I don't follow why the headers of sections are marked as skipped. I would expect for them to be in the `headedBySectionSteps` list.

Maybe the step representing the header can also be marked as active/completed/pending=default based on the following rules:

* if the current step is of the section, then the section is active (note, the question step itself will also be active)
* if the current step is NOT of the section, and there are already answers within the section, then section is completed.
* otherwise, is pending=default.

Between these changes and knowing the step type, we can tweak later the styling to show progress in sections nicely I think. Again, I don't mind the styling in this PR.

Reply:
I think this problem was solved in [this commit](https://github.com/instedd/surveda/pull/1698/commits/f9981048f94c70b64358c328fee36bb7d2d9c489)